src/Common/MemoryTracker.cpp

```diff
@@ -533,7 +539,11 @@ void MemoryTracker::updateAllocated(Int64 allocated_, bool log_change)
             "Correcting the value of global memory tracker from {} to {}",
             ReadableSize(total_memory_tracker.amount.load(std::memory_order_relaxed)),
             ReadableSize(allocated_));
-    total_memory_tracker.amount.store(new_amount, std::memory_order_relaxed);
+
+    auto current_amount = total_memory_tracker.amount.exchange(new_amount, std::memory_order_relaxed);
+    total_memory_tracker.uncorrected_amount += (current_amount - total_memory_tracker.last_corrected_amount);
```

Comment:
In case auto correction will be enabled (and from one side it is good to do, to avoid extra reported issues, but from other - it may decrease priority of this memory tracking issue) - https://github.com/ClickHouse/ClickHouse/pull/75714, this will not be correct, since there will be always some drift, right?

Reply:
Can you explain a bit more?
I calculate the deltas between current value and last set value so `uncorrected_amount` should contain value which would be stored as if `memory_worker_correct_memory_tracker` is disabled.
Maybe I'm mathing wrong something here.